pkg/controller/apmserver/certificates/reconcile.go

```diff
@@ -48,15 +49,15 @@ func Reconcile(
 		as,
 		labels,
 		certificates.HTTPCAType,
-		rotation,
+		caRotation,
 	)
 	if err != nil {
 		return results.WithError(err)
 	}
 
 	// handle CA expiry via requeue
 	results.WithResult(reconcile.Result{
-		RequeueAfter: certificates.ShouldRotateIn(time.Now(), httpCa.Cert.NotAfter, rotation.RotateBefore),
+		RequeueAfter: certificates.ShouldRotateIn(time.Now(), httpCa.Cert.NotAfter, caRotation.RotateBefore),
```

Comment:
Should we also make that call for the other certs?

Reply:
I am not sure I follow your comment here @sebgl. Do you mean we should requeue within the validity period of the individual certificates and not just before the CA expires?

Reply:
Yeah I mean also requeueing after we reconcile HTTP certificates below. Since we have `caRotation` and `certRotation` that can be different, it would make sense to make sure we requeue before any of these are reached? Maybe I'm missing something here.

Reply:
Should we not requeue on the minimum of the two?

Reply:
Yes! That's why I thought we would call `results.WithResult ()` twice (one for CA, one for cert), and then we let the results aggregation do its job of picking the most appropriate requeue?

Reply:
SGTM